Core/Core/API/API.swift

```diff
@@ -190,18 +145,54 @@ public class URLSessionAPI: API {
     }
 
     private func flushRefreshQueue() {
-        let tasks = refreshQueue
-        tasks.forEach { $0() }
+        for task in refreshQueue { task() }
```

Comment:
We should iterate over a copy so as to avoid race conditions where we are appending and iterating over the same list. I remember seeing a crash on it once.

Reply:
In fact, we get crashes on append too quite a bit according to crash reports. I think we should switch to using OperationQueue for the refresh queue rather than an array. But we can do that separately.